src/sentry/tasks/embeddings_grouping/utils.py

```diff
@@ -94,27 +95,26 @@ def filter_snuba_results(snuba_results, groups_to_backfill_with_no_embedding, pr
 def initialize_backfill(
     project_id: int,
     cohort: str | list[int] | None,
-    last_processed_group_index: int | None,
+    last_processed_group_id: int | None,
     last_processed_project_index: int | None,
     redis_client: StrictRedis | RedisCluster,
 ):
     logger.info(
         "backfill_seer_grouping_records.start",
         extra={
             "project_id": project_id,
-            "last_processed_index": last_processed_group_index,
+            "last_processed_index": last_processed_group_id,
         },
     )
     project = Project.objects.get_from_cache(id=project_id)
     if not features.has("projects:similarity-embeddings-backfill", project):
         raise FeatureError("Project does not have feature")
 
-    if last_processed_group_index is None:
-        last_processed_group_index_ret = int(
-            redis_client.get(make_backfill_grouping_index_redis_key(project_id)) or 0
-        )
+    last_processed_group_id_ret = redis_client.get(make_backfill_grouping_id_redis_key(project_id))
+    if last_processed_group_id is None and last_processed_group_id_ret is not None:
+        last_processed_group_id_ret = int(last_processed_group_id_ret)
     else:
-        last_processed_group_index_ret = last_processed_group_index
+        last_processed_group_id_ret = last_processed_group_id
```

Comment:
Probably-for-a-different-PR nit: 

If I understand correctly what's happening in these lines, I think this would be easier to grock like this:

(it won't let me use the suggestion function for some reason)

```python
    last_processed_group_id = last_processed_group_id or int(
        redis_client.get(make_backfill_grouping_id_redis_key(project_id))
    )
```

and then just return `last_processed_group_id`.